app/templates/components/public/ticket-list.hbs

```diff
@@ -58,10 +58,30 @@
         </tr>
       </tfoot>
     </table>
-    <div class="ui row grid">
-      <div class="column right aligned">
-        <a href="#">{{t 'Enter promotional code'}}</a>
-      </div>
+    <div class="ui grid">
+      {{#if enterPromotionalCode}}
+        <div class="ui row">
+          <div class="{{if device.isBiggerThanTablet 'right floated eight wide'}} column">
+            <div class="field">
+              <div class="ui action fluid input">
+                {{input type='text' name='promotional_code'}}
+                <div role='button' class="ui icon button" {{action 'applyPromotionalCode'}}>
+                  <i class="checkmark icon"></i>
+                </div>
+                <div role='button' class="ui black icon button" {{action 'togglePromotionalCode'}}>
+                  <i class="remove icon"></i>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      {{else}}
+        <div class="ui row">
+          <div class="column right aligned">
+            <a href="#" {{action 'togglePromotionalCode'}}>{{t 'Enter promotional code'}}</a>
+          </div>
+        </div>
+      {{/if}}
     </div>
     <div class="ui row">
       <div class="ui grid">
```

Comment:
@CosmicCoder96 content of this grid seems to be shifted to left should be center aligned 
![image](https://user-images.githubusercontent.com/20153241/28042287-3f8fcbf4-65ea-11e7-99a2-09d077e8f72d.png)